### ISSUE TITLE
AutoDetect indentation not working as expected

### ISSUE THREAD

**Issue by gsmet** [openrewrite/rewrite#1498]
Hi,

Still making progress on usage of OpenRewrite in Quarkus and I hit an issue that is minor but quite annoying as it makes our pom files all weird: autodetecting the indentation leads to weird results.
For instance, for the very same run and the very same file, I end up with exotic and different values (I added some logging here: https://github.com/openrewrite/rewrite/blob/main/rewrite-xml/src/main/java/org/openrewrite/xml/style/Autodetect.java#L89 ):
```
independent-projects/bootstrap/bom/pom.xml
Indent: 12
independent-projects/bootstrap/bom/pom.xml
Indent: 12
independent-projects/bootstrap/bom/pom.xml
Indent: 12
independent-projects/bootstrap/bom/pom.xml
Indent: 16
```
I could understand it being exotic but it being non deterministic is a bit weird.

The indent they should find is `4` and not `12`.
The file that is modified is available here: https://github.com/quarkusio/quarkus/blob/main/independent-projects/bootstrap/bom/pom.xml .

Here is our recipe file as it might help to see the rules that trigger auto detection (with a txt extension as it's not possible to upload yml files):
[rewrite.txt](https://github.com/openrewrite/rewrite/files/8341917/rewrite.txt)

I think I would go with a slightly different logic and only consider the indentation of elements of level 1 from the root to detect that but AFAICS the depth is not available in the visitor.

Unfortunately, I haven't found a way to apply a custom style as styles defined in a local `rewrite.yml` are not taken into account when adjusting the pom file with the Maven plugin.

For now, I will harcode the value in our fork.

Thanks!

**Comment by pway99** [comment]
thanks for the issue and PR, we'll have a look.

**Comment by traceyyoshima** [comment]
Hi @gsmet, thanks for reporting the issue!

I think the depth is available but requires a slightly different approach.

The XML spec defines a document as `document ::= prolog element misc*`.
So, an `Xml.Document` will contain a `Tag root` and the root will contain a `List<Content>`.

Each of the prefixes in the `List<Content>` will be the first-level prefix unless the root is indented.
To visit the next level, filter the list for tags and check the prefixes of their `List<Content>`

Note that it's important to filter out non-tag content or at least comments:
`content.stream().filter(c -> c instanceOf Xml.Tag).map(c -> (Xml.Tag) c).collect ... `

**Comment by gsmet** [comment]
@traceyyoshima I went with a slightly different approach in my PR here https://github.com/openrewrite/rewrite/pull/1504 and it gave some interesting results.

**Comment by traceyyoshima** [comment]
@gsmet I haven't had a chance to look at the PR or the recent addition of XML Autoformatting. I think @pway99 will be looking through the PR tomorrow.

Autodetecting formats can be pretty tricky since there are so many different approaches.
I.E. 3 spaces, or several spaces and a tab. Unfortunately, mixed tabs and spaces do exist.

**Comment by gsmet** [comment]
So 3 spaces would work OK with the new method I suggested. Mixed tabs and spaces wouldn't work but it didn't really work well before either as you counted spaces and tabs in the same way and switches to one or other depending of their respective number.

**Comment by jkschneider** [comment]
@gsmet Just as an update -- I think we just need to limit the indentation calculation to prefixes on tags right now, since attributes can be continuation indented and therefore may mess up this calculation in some situations. We can do this work if that's helpful.

Haven't had a chance to thank you for working to improve the algorithm. We'll get it in there shortly one way or another!

**Comment by gsmet** [comment]
@jkschneider I don’t think it will solve my issue, given the document I have. 
Considering only level 1 tags would work but attributes don’t play a role in my issue. 
My problem is that I have more lines indented with 12 spaces than four. So if you just consider frequency without considering the level of the tags (or the approach I used), you end up considering 12 as the indent. And just considering the tags won’t change that if you don’t limit this to level 1.

**Comment by gsmet** [comment]
@jkschneider I will add a test to my PR so that you can see what's going on in our case. I'm not saying what I did is the right fix but with a test, it will be easier to discuss.

Need to make some progress on our migration first but I'll get to it tomorrow.

**Comment by pway99** [comment]
Hello @gsmet,  I will be focusing on helping you out with this PR, just added a test cc6a8635df987a988cefd2152be4c8aa68baef7b as a start.

**Comment by jkschneider** [comment]
Fixed with 05a9714da03f26d9064c4b05a1e01b8527f9854f and e1445e8e1c87f233db1595b5339ebbeaebef86ce.
